runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/PlayerComparisonOverlay.java

```diff
@@ -0,0 +1,190 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.opponentinfo;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import javax.inject.Inject;
+import net.runelite.api.Actor;
+import net.runelite.api.Client;
+import net.runelite.api.Player;
+import net.runelite.api.Skill;
+import net.runelite.client.game.HiscoreManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.LineComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.components.TitleComponent;
+import net.runelite.client.util.Text;
+import net.runelite.http.api.hiscore.HiscoreResult;
+import net.runelite.http.api.hiscore.HiscoreSkill;
+
+class PlayerComparisonOverlay extends Overlay
+{
+	private static final Color HIGHER_STAT_TEXT_COLOR = Color.GREEN;
+	private static final Color LOWER_STAT_TEXT_COLOR = Color.RED;
+	private static final Color NEUTRAL_TEXT_COLOR = Color.WHITE;
+	private static final Color HIGHLIGHT_COLOR = new Color(255, 200, 0, 255);
+
+	private static final Skill[] COMBAT_SKILLS = new Skill[]{
+		Skill.ATTACK,
+		Skill.STRENGTH,
+		Skill.DEFENCE,
+		Skill.HITPOINTS,
+		Skill.RANGED,
+		Skill.MAGIC,
+		Skill.PRAYER
+	};
+
+	private static final HiscoreSkill[] HISCORE_COMBAT_SKILLS = new HiscoreSkill[]{
+		HiscoreSkill.ATTACK,
+		HiscoreSkill.STRENGTH,
+		HiscoreSkill.DEFENCE,
+		HiscoreSkill.HITPOINTS,
+		HiscoreSkill.RANGED,
+		HiscoreSkill.MAGIC,
+		HiscoreSkill.PRAYER
+	};
+
+	private static final String LEFT_COLUMN_HEADER = "Skill";
+	private static final String RIGHT_COLUMN_HEADER = "You/Them";
+
+	private final Client client;
+	private final OpponentInfoPlugin opponentInfoPlugin;
+	private final OpponentInfoConfig config;
+	private final HiscoreManager hiscoreManager;
+	private final PanelComponent panelComponent = new PanelComponent();
+
+	private HiscoreResult lastOpponentSkills;
```

Comment:
What is the point of this variable? you are doing lookup every render cycle. Just send the result directly to the method that creates the table and remove this, or try to not do lookups every frame to make use of this

Reply:
Because of the async lookup, there will always be a short period of time where the first time a player is interacted with, they will not have any stats to display. This allows the table to show a previous interacted player's stats during that time.

Reply:
But why would you want to show previous interacted player stats? I would just follow what opponentInfoOverlay does atm and just show the data when they are available

Reply:
Fixed by showing a "loading" table instead of previous opponent's stats.